builtin/providers/google/provider.go

```diff
@@ -62,3 +67,28 @@ func providerConfigure(d *schema.ResourceData) (interface{}, error) {
 
 	return &config, nil
 }
+
+func validateAccountFile(v interface{}, k string) (warnings []string, errors []error) {
+	value := v.(string)
+
+	if value == "" {
+		return
+	}
+
+	var account accountFile
+	if err := json.Unmarshal([]byte(value), &account); err != nil {
```

Comment:
we need to detect if `value` here is a JSON file, or the contents of a file. If I supply `somefile.json` or even just the word `what`, this will throw an error:

```
invalid character 'w' looking for beginning of value
```

I'm thinking we shouldn't return in this error block, instead return if there is no error (we found json, yay)